### ISSUE TITLE
NullPointerException when try to parse dot graph with empty attributes

### ISSUE THREAD

**Issue by Schalli1987** [jgrapht/jgrapht#746]
```
 * JGraphT version: 1.3.0
 * Java version (java -version)/platform:  java version "1.8.0_141"
Java(TM) SE Runtime Environment (build 1.8.0_141-b15)
Java HotSpot(TM) 64-Bit Server VM (build 25.141-b15, mixed mode)
```

**Issue**
DOTImporter throws NullPointerException when trying to parse a vertex with empty attributes.

**Steps to reproduce (small coding example)**
```java
@Test
public void simpleGraphTest() throws Exception {
    String sG = "digraph g {\n" + // gets parsed correctly
            "name=\"TEST\";\n" + // gets parsed correctly
            "\n" + // gets parsed correctly
            "graph[]\n" + // gets parsed correctly
            "\n" + // gets parsed correctly
            "  anyNodeWithAttributes [color=\"red\"];\n" + // gets parsed correctly
            "  anyNodeWithoutAttributes;\n" + // gets parsed correctly
            "  anyNodeWithEmptyAttributes [];\n" + // NPE!
            "\n" + //
            "}";
    VertexProvider<String> vertexProvider = (node, attributes) -> {
        return node;
    };
    EdgeProvider<String, DefaultEdge> edgeProvider = (from, to, label, attributes) -> {
        return new DefaultEdge();
    };
    final DOTImporter<String, DefaultEdge> dotImporter = new DOTImporter<>(vertexProvider, edgeProvider);
    Graph<String, DefaultEdge> sGraph = new DefaultDirectedGraph<>(DefaultEdge.class);
    dotImporter.importGraph(sGraph, new ByteArrayInputStream(sG.getBytes()));
}
```

**Expected behaviour**
Graph should be imported correctly, thus the empty attributes should be parsed like no attributes were given.

**Comment by d-michail** [comment]
Thanks!

I opened up a pull-request with a fix. Let us know if you encounter any more bugs.

**Comment by jkinable** [comment]
Thanks for reporting! Bug has been addressed in #747
